src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.EcKey.cs

```diff
@@ -25,7 +24,49 @@ internal static partial class Crypto
         [return: MarshalAs(UnmanagedType.Bool)]
         internal static extern bool EcKeyUpRef(IntPtr r);
 
-        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_EcKeyGetCurveName")]
-        internal static extern int EcKeyGetCurveName(SafeEcKeyHandle ecKey);
+        [DllImport(Libraries.CryptoNative)]
+        private static extern int CryptoNative_EcKeyGetSize(SafeEcKeyHandle ecKey, out int keySize);
+        internal static int EcKeyGetSize(SafeEcKeyHandle key)
+        {
+            int keySize;
+            int rc = CryptoNative_EcKeyGetSize(key, out keySize);
+            if (rc == 1)
+            {
+                return keySize;
+            }
+            throw Interop.Crypto.CreateOpenSslCryptographicException();
+        }
+
+        [DllImport(Libraries.CryptoNative)]
+        private static extern int CryptoNative_EcKeyGetOidValueCurveName(SafeEcKeyHandle ecKey, out int nid);
+
+        internal static string EcKeyGetOidValueCurveName(SafeEcKeyHandle key)
+        {
+            int nidCurveName;
+            int rc = CryptoNative_EcKeyGetOidValueCurveName(key, out nidCurveName);
+            if (rc == 1)
+            {
+                if (nidCurveName == Interop.Crypto.NID_undef)
+                    return string.Empty;
```

Comment:
If something is wrong with the key - key is invalid or doesn't have a curve set yet. Not possible in normal flows. Will add assert to make it more clear this is not expected